programs/bpf_loader/src/syscalls.rs

```diff
@@ -515,19 +515,31 @@ fn translate_string_and_do(
 
 type SyscallContext<'a, 'b> = Rc<RefCell<&'a mut InvokeContext<'b>>>;
 
-/// Abort syscall functions, called when the BPF program calls `abort()`
-/// LLVM will insert calls to `abort()` if it detects an untenable situation,
-/// `abort()` is not intended to be called explicitly by the program.
-/// Causes the BPF program to be halted immediately
-pub struct SyscallAbort<'a, 'b> {
-    invoke_context: SyscallContext<'a, 'b>,
-}
-impl<'a, 'b> SyscallAbort<'a, 'b> {
-    pub fn init(invoke_context: SyscallContext<'a, 'b>) -> Box<(dyn SyscallObject<BpfError> + 'a)> {
-        Box::new(Self { invoke_context })
-    }
+macro_rules! declare_syscall {
+    ($(#[$attr:meta])* $name:ident, $call:item) => {
+        $(#[$attr])*
```

Comment:
Is there a use case for the meta attributes or is it just in case we want to add it later?

Reply:
The meta attributes pick up the doc comments, we could leave them out and just document above the macros calls since these structs are no longer public